packages/components/src/modal/style.scss

```diff
@@ -146,4 +146,25 @@
 		outline: 2px solid transparent;
 		outline-offset: -2px;
 	}
+
+	@include break-small() {
+		&.has-width-small,
+		.has-width-medium,
+		.has-width-large {
+			width: 100%;
+		}
+
+		// The following widths were selected to align with existing baselines
+		// found elsewhere in the editor.
+		// See https://github.com/WordPress/gutenberg/pull/54471#issuecomment-1723818809
+		&.has-width-small {
+			max-width: 384px;
+		}
+		&.has-width-medium {
+			max-width: 512px;
+		}
+		&.has-width-large {
+			max-width: 832px;
+		}
```

Comment:
Should these values be abstracted into variables? It may be useful for container widths (see https://github.com/WordPress/gutenberg/issues/53322).

Reply:
> Should these values be abstracted into variables?

If we want these values to be shared across the whole project (not just modal / components), we could for now add the to the base styles package (knowing that they could be later be moved to whatever the centralised place for style variables is going to be)